src/shared/sysctl-util.c

```diff
@@ -53,14 +57,18 @@ char *sysctl_normalize(char *s) {
 
 int sysctl_write(const char *property, const char *value) {
         char *p;
+        _cleanup_close_ int fd = -1;
 
         assert(property);
         assert(value);
 
         log_debug("Setting '%s' to '%s'", property, value);
 
         p = strjoina("/proc/sys/", property);
-        return write_string_file(p, value, WRITE_STRING_FILE_DISABLE_BUFFER);
+        fd = open(p, O_WRONLY);
+        if (fd == -1)
```

Comment:
irrelevant nitpick: we usually check for syscall return value errors like this with < 0, not == -1...